nav2_regulated_pure_pursuit_controller/include/nav2_regulated_pure_pursuit_controller/regulated_pure_pursuit_controller.hpp

```diff
@@ -208,6 +208,21 @@ class RegulatedPurePursuitController : public nav2_core::Controller
    */
   geometry_msgs::msg::PoseStamped getLookAheadPoint(const double &, const nav_msgs::msg::Path &);
 
+  /**
+   * @brief checks for the cusp position
+   * @param pose Pose input to determine the cusp position
+   * @return robot distance from the cusp
+   */
+  double detectPathOrientation(const geometry_msgs::msg::PoseStamped & pose);
+
+  /**
+   * @brief determines the dot product
+   * @param u_x, u_y, represents the vector u
+   * @param v_x, v_y, represents the vector v 
+   * @return robot distance from the cusp
+   */
+  inline double dot(double u_x, double u_y, double v_x, double v_y);
```

Comment:
Please remove this function. You only use it once and it is so simple you can just do it inline in the actual function that uses it. I think I made this comment too last time :laughing:

Reply:
Lol, I thought, you wanted to use "inline" function!